.circleci/config.yml

```diff
@@ -74,8 +74,9 @@ jobs:
       - run:
           name: setup Python venv
           command: |
-            pip install numpy==1.23.5 cython!=3.0.3 pybind11 pythran ninja meson
+            pip install numpy cython!=3.0.3 pybind11 pythran ninja meson
```

Comment:
Unpinning NumPy and/or bumping the Python version may be related to the new refguide-check failures, but I don't mind putting this in since the `myst-nb` pin is needed for now.

You may want to keep an eye on the CI and i.e., re-pin NumPy if that helps with the refguide failures. One nice thing about pinning near the oldest supported NumPy/Python versions is that it mirrors how I build docs at release time, but that's mostly a convenience thing rather than requirement for doc build proper.

Reply:
Oh got it, I wasn't sure what the purpose of the pin was. Thanks!